Index: webkit/glue/webframe.h
===================================================================
--- webkit/glue/webframe.h	(revision 5665)
+++ webkit/glue/webframe.h	(working copy)
@@ -364,6 +364,8 @@
   // Only for test_shell
   virtual int PendingFrameUnloadEventCount() const = 0;
 
+  virtual void SetAllowsScrolling(bool allows_scrolling) = 0;
+
  private:
   DISALLOW_COPY_AND_ASSIGN(WebFrame);
 };
Index: webkit/glue/webframe_impl.h
===================================================================
--- webkit/glue/webframe_impl.h	(revision 5665)
+++ webkit/glue/webframe_impl.h	(working copy)
@@ -264,7 +264,7 @@
   // Sets whether the WebFrameImpl allows its document to be scrolled.
   // If the parameter is true, allow the document to be scrolled.
   // Otherwise, disallow scrolling
-  void SetAllowsScrolling(bool flag);
+  virtual void SetAllowsScrolling(bool flag);
 
   // Returns true if the frame CSS is in "printing" mode.
   bool printing() const { return printing_; }
