--- jsoncpp.orig/src/lib_json/json_internalarray.inl	2006-06-26 19:52:43.000000000 -0700
+++ json_internalarray.inl.h	2006-06-26 17:35:42.000000000 -0700
@@ -111,8 +111,10 @@
       if ( minNewIndexCount > newIndexCount )
          newIndexCount = minNewIndexCount;
       void *newIndexes = realloc( indexes, sizeof(Value*) * newIndexCount );
-      if ( !newIndexes )
-         throw std::bad_alloc();
+      // Changed by girts:
+      // if ( !newIndexes )
+      //   throw std::bad_alloc();
+      assert(newIndexes);
       indexCount = newIndexCount;
       indexes = static_cast<Value **>( newIndexes );
    }
@@ -252,8 +254,8 @@
 
 ValueInternalArray::ValueInternalArray( const ValueInternalArray &other )
    : pages_( 0 )
-   , pageCount_( 0 )
    , size_( other.size_ )
+   , pageCount_( 0 )
 {
    PageIndex minNewPages = other.size_ / itemsPerPage;
    arrayAllocator()->reallocateArrayPageIndex( pages_, pageCount_, minNewPages );
