--- jsoncpp.orig/src/lib_json/json_writer.cpp	2006-06-26 19:52:43.000000000 -0700
+++ json_writer.cc	2006-06-26 17:10:08.000000000 -0700
@@ -1,4 +1,4 @@
-#include <json/writer.h>
+#include "third_party/jsoncpp/writer.h"
 #include <utility>
 #include <assert.h>
 #include <stdio.h>
@@ -13,7 +13,8 @@
                           char *&current )
 {
    *--current = 0;
-   char *end = current;
+   // Commented out by girts to fix unused variable warning:
+   // char *end = current;
    do
    {
       *--current = (value % 10) + '0';
