--- jsoncpp.orig/src/lib_json/json_batchallocator.h	2006-06-26 19:52:43.000000000 -0700
+++ json_batchallocator.h	2006-06-26 18:04:29.000000000 -0700
@@ -25,11 +25,13 @@
 class BatchAllocator
 {
 public:
-   typedef AllocatedType AllocatedType;
+   // girts: the compiler does not like the typedef, seems to work fine
+   // without one.
+   // typedef AllocatedType AllocatedType;
 
    BatchAllocator( unsigned int objectsPerPage = 255 )
-      : objectsPerPage_( objectsPerPage )
-      , freeHead_( 0 )
+      : freeHead_( 0 )
+      , objectsPerPage_( objectsPerPage )
    {
 //      printf( "Size: %d => %s\n", sizeof(AllocatedType), typeid(AllocatedType).name() );
       assert( sizeof(AllocatedType) * objectPerAllocation >= sizeof(AllocatedType *) ); // We must be able to store a slist in the object free space.
@@ -122,4 +124,3 @@
 # endif // ifndef JSONCPP_DOC_INCLUDE_IMPLEMENTATION
 
 #endif // JSONCPP_BATCHALLOCATOR_H_INCLUDED
-
